core/kernel/notif.c

```diff
@@ -106,6 +123,273 @@ void notif_send_async(uint32_t value)
 	cpu_spin_unlock_xrestore(&notif_lock, old_itr_status);
 }
 
+#if defined(CFG_CORE_IT_NOTIF)
+/*
+ * Retrieve an pending interrupt even and whether other are pending
+ *
+ * @it_valid (Output) true if returning a pending interrupt ID, false otherwise
+ * @it_pending (Output) true if other interrupts are pending, false otherwise
+ * Return the ID of a pending interrupt when output @*it_valid is true
+ *
+ * This function is callback from a fastcall context.
+ */
+uint32_t notif_it_get_value(bool *it_valid, bool *it_pending)
+{
+	const int max_bit = NOTIF_IT_VALUE_MAX + 1;
+	uint32_t old_itr_status = 0;
+	uint32_t res = 0;
+	int bit = 0;
+
+	old_itr_status = cpu_spin_lock_xsave(&it_lock);
+
+	/* Get 1st pending non masked interrupt */
+	bit = -1;
+	do {
+		bit_ffs_from(notif_it_pending, max_bit, bit + 1, &bit);
+	} while (bit >= 0 && bit_test(notif_it_masked, bit));
+
+	*it_valid = (bit >= 0);
+	if (!*it_valid) {
+		*it_pending = false;
+		goto out;
+	}
+
+	/* Clear pending interrupt from the record */
+	res = bit;
+	bit_clear(notif_it_pending, res);
+	bit_ffs(notif_it_pending, max_bit, &bit);
+
+	/* Check if there is  another pending non masked interrupt */
+	bit = -1;
+	do {
+		bit_ffs_from(notif_it_pending, max_bit, bit + 1, &bit);
+	} while (bit >= 0 && bit_test(notif_it_masked, bit));
+	*it_pending = (bit >= 0);
+
+out:
+	cpu_spin_unlock_xrestore(&it_lock, old_itr_status);
+
+	return res;
+}
+
+/* This function can execute in a fastcall interrupt masked context */
+static void set_it_mask(unsigned int id, bool do_mask)
```

Comment:
We have "id", "it value", and "it number" for the same thing.  I'd prefer if we could stick to one name, I think "it number" is the clearest.

Reply:
ok `itr_num` here and there?